cmd/bosun/conf/conf.go

```diff
@@ -865,20 +871,110 @@ func (c *Conf) loadAlert(s *parse.SectionNode) {
 		}
 	}
 	a.returnType = ret
+	c.findAllDependencies(&a)
 	c.Alerts[name] = &a
 	c.OrderedAlerts = append(c.OrderedAlerts, &a)
 }
 
+func (c *Conf) findAllDependencies(a *Alert) {
+	//Notifications and lookups are dependencies
+	var walkNotifications = func(n *Notifications) {
+		for _, l := range n.Lookups {
+			a.Dependencies = append(a.Dependencies, &l.ConfItem)
```

Comment:
Instead of being forced to do stuff like this (where a function dealing with Alerts has to know about looks and chains in notifications), what if we had a dependency interface, that each type (alert, lookup, notification, macro, template) implemented?